services/src/datasets/external/edr.rs

```diff
@@ -0,0 +1,1241 @@
+        let temporal_extent = self.extent.temporal.as_ref().ok_or_else(|| {
+            geoengine_operators::error::Error::DatasetMetaData {
+                source: Box::new(EdrProviderError::MissingTemporalExtent),
+            }
+        })?;
+
+        Ok(TimeInterval::new_unchecked(
+            TimeInstance::from_str(&temporal_extent.interval[0][0]).unwrap(),
+            TimeInstance::from_str(&temporal_extent.interval[0][1]).unwrap(),
+        ))
+    }
+
+    fn get_bounding_box(&self) -> Result<BoundingBox2D, geoengine_operators::error::Error> {
+        let spatial_extent = self.extent.spatial.as_ref().ok_or_else(|| {
+            geoengine_operators::error::Error::DatasetMetaData {
+                source: Box::new(EdrProviderError::MissingSpatialExtent),
+            }
+        })?;
+
+        Ok(BoundingBox2D::new_unchecked(
+            Coordinate2D::new(spatial_extent.bbox[0][0], spatial_extent.bbox[0][1]),
+            Coordinate2D::new(spatial_extent.bbox[0][2], spatial_extent.bbox[0][3]),
+        ))
+    }
+
+    fn select_output_format(&self) -> Result<String, geoengine_operators::error::Error> {
+        for format in &self.output_formats {
+            if GEO_FILETYPES
+                .get_or_init(init_geo_filetypes)
+                .contains_key(format)
+            {
+                return Ok(format.to_string());
+            }
+        }
+        Err(geoengine_operators::error::Error::DatasetMetaData {
+            source: Box::new(EdrProviderError::NoSupportedOutputFormat),
+        })
+    }
+
+    fn is_raster_file(&self) -> Result<bool, geoengine_operators::error::Error> {
+        Ok(*GEO_FILETYPES
+            .get_or_init(init_geo_filetypes)
+            .get(&self.select_output_format()?)
+            .expect("can only return values in map"))
+    }
+
+    fn get_vector_download_url(
+        &self,
+        base_url: &Url,
+        height: &str,
+        discrete_vrs: &[String],
+    ) -> Result<(String, String), geoengine_operators::error::Error> {
+        let spatial_extent = self.extent.spatial.as_ref().ok_or_else(|| {
+            geoengine_operators::error::Error::DatasetMetaData {
+                source: Box::new(EdrProviderError::MissingSpatialExtent),
+            }
+        })?;
+        let temporal_extent = self.extent.temporal.as_ref().ok_or_else(|| {
+            geoengine_operators::error::Error::DatasetMetaData {
+                source: Box::new(EdrProviderError::MissingTemporalExtent),
+            }
+        })?;
+        let z = if height == "default" {
+            String::new()
+        } else if self.extent.has_discrete_vertical_axis(discrete_vrs) {
+            format!("&z={height}")
+        } else {
+            format!("&z={height}%2F{height}")
+        };
+        let download_url = format!(
+            "/vsicurl_streaming/{}collections/{}/cube?bbox={},{},{},{}{}&datetime={}%2F{}&f={}",
```

Comment:
Two things I stumbled over:
- This is missing parameter-names.
- The z parameter does not have a &z prefix. Also in the raster download url. Is this by design?